frontend/src/lib/utils/eventUsageLogic.ts

```diff
@@ -102,6 +103,20 @@ function hasGroupProperties(properties: AnyPropertyFilter[] | PropertyGroupFilte
     return !!flattenedProperties && flattenedProperties.some((property) => property.group_type_index != undefined)
 }
 
+function usedCohortFilterIds(properties: AnyPropertyFilter[] | PropertyGroupFilter | undefined): PropertyFilterValue[] {
+    const flattenedProperties = convertPropertyGroupToProperties(properties)
+    const cohortIds: PropertyFilterValue[] = []
+    if (flattenedProperties) {
+        for (const property of flattenedProperties) {
+            if (property.type === 'cohort') {
+                cohortIds.push(property.value)
+            }
+        }
+    }
```

Comment:
Let's avoid for loops in JS—something like this could work

```
const cohortIds = flattenedProperties.filter(p => p.type === 'cohort').map(p => p.value)
```

Reply:
stylistic question here. if I declare the const inside `if (flattenedProperties)`, how do i avoid using a mutable to return at the end of the function or would you return directly inside the if condition